### PR TITLE
interp: add -s support for read builtin

### PR DESCRIPTION
**Pull request by davidhulick**
This implements the -s option for the read builtin using the ReadPassword function in the golang.org/x/term package. This is a less featureful (and as a result, simpler) implementation of the -s flag than is present in https://github.com/mvdan/sh/pull/865

Like PR865, I am also unsure of how to test the -s flag since it requires interacting with the terminal input.

### CODE REVIEW

File: interp/builtin.go [hunk 1]
```diff
@@ -559,10 +561,15 @@ func (r *Runner) builtinCode(ctx context.Context, pos syntax.Pos, name string, a
 		r.setErr(returnStatus(code))
 	case "read":
 		var prompt string
+		var line []byte
+		var err error
```
Comment (mvdan) [inline interp/builtin.go:565]: keep these vars declared further below, to ont keep as much in scope at once.

File: interp/builtin.go [hunk 2]
```diff
@@ -589,7 +596,12 @@ func (r *Runner) builtinCode(ctx context.Context, pos syntax.Pos, name string, a
 			r.out(prompt)
 		}
 
-		line, err := r.readLine(raw)
+		if silent {
+			line, err = term.ReadPassword(int(syscall.Stdin))
+		}else{
```
Comment (mvdan) [inline interp/builtin.go:601]: did you run gofmt?

**Review by mvdan — APPROVED** [review]
Thanks! SGTM, just two nits. I agree testing this is hard and not really worth it, given that it's really three lines of simple code.

File: interp/builtin.go (hunk 1 above)
Comment (davidhulick) [reply interp/builtin.go]: Moved these variable declarations further down.

File: interp/builtin.go (hunk 2 above)
Comment (davidhulick) [reply interp/builtin.go]: My apologies! Fixed.

**Pull request merged**